hefquin-engine/src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/PhysicalPlanFactory.java

```diff
@@ -396,7 +396,7 @@ public static PhysicalPlan createPlan( final BinaryLogicalOp rootOp,
 	                                       final QueryPlanningInfo qpInfo,
 	                                       final PhysicalPlan subplan1,
 	                                       final PhysicalPlan subplan2 ) {
-		final BinaryPhysicalOp pop = LogicalToPhysicalOpConverter.convert(rootOp);
+		final BinaryPhysicalOp pop = (BinaryPhysicalOp) LogicalToPhysicalOpConverter.convert(rootOp);
```

Comment:
This doesn't seem right. For binary operators we need to be able to check whether `subplan1.getExpectedVariables()` and `subplan2.getExpectedVariables()` can be used. To this end, it is actually necessary to change `PhysicalOpFactory#supports` to have the following signature:
```java
boolean supports( LogicalOperator lop, ExpectedVariables ... inputVars );
```
And likewise for `PhysicalOpRegistry#create`.

Reply:
There are a few cases when we can explicitly create plans with a particular physical operator (e.g. `PhysicalPlanFactory#createPlanWithUnion` and `PhysicalPlanFactory#createPlanWithHashJoin`). Do we also want to check that the physical operator is supported in those cases or would this be the responsibility of the calling function?

Also, is it correct to assume that we would like to add the requirement that at least one variables is shared for `PhysicalOpHashJoin, `PhysicalOpNaiveNestedLoopsJoin`, and `PhysicalOpSymmetricHashJoin`? Which one should fallback/degrade to Cartesian product?